Adds instructions to the Webots guide for recording a full game (#184)
NUbots/NUbook: src/book/03-guides/02-tools/03-webots.mdx

```

Where `<name of controller>` is the name you want to use for the new controller. A folder and files will be created with this name, so they must be valid file names (don't use spaces in your controller name).

## Recording a Game

NUbots runs full Webots games to benchmark the current performance of the code. The following steps describe how to create a video of a full game.

1. Ensure you have followed the previous steps [Install Webots](#install-webots) and [Running the Official RoboCup Simulation in Webots](#running-the-official-robocup-simulation-in-webots).

2. Set up the game configuration.

   1. Open the [game.json file](https://github.com/NUbots/NUWebots/blob/main/webots/projects/samples/contests/robocup/controllers/referee/game.json).
   2. Change the `"blue":` `"id"` to `12`.

>    3. Change the `"blue":` `config:` field to `"team_1.json"`.

Reviewer: ```suggestion
   3. Change the `"blue":` `"config"` field to `"team_1.json"`.
```

>    At the moment, there is an issue due to the fact that both teams have the same team ID since they are both NUbots teams. A penalty to one robot will also penalise the corresponding robot on the other team.

Reviewer: This deserves an issue on NUbots IMO. We should be trying to fix this

Reviewer 2: Do we just need to change the team_id [here](https://github.com/NUbots/NUbots/blob/main/module/support/configuration/GlobalConfig/data/config/GlobalConfig.yaml#L2) to sort this?

Author: Yep, that just needs to be different for the robots on different teams.

3. Open a terminal and run the following to open Webots with the Robocup world and the GameController:

   ```bash

>    $WEBOTS_HOME/webots $WEBOTS_HOME/projects/samples/contests/robocup/worlds/robocup.wbt

Reviewer: Do we need the normal environment variables defined in line, like we usually have when running webots?

Author: Which environment variables? I always just run `./webots` and it all works fine, unless I'm adding in the game controller filter.

Reviewer: Looking at the webots guide now, I was running on outdated info. I was thinking of the variables we add to the bashrc and the UDP_FILTER - I always passed those to the webots call directly. All good

   ```

   Pause the simulation when it opens. The world and robots will load in but the game will not start yet.

4. Open a separate terminal and run the robot code.

   ```bash
   ./b webots game -r 6 webotsrobocup

>    ```

Reviewer 2: Although this will throw an error asking the user to build the image if it hasnt been, its probably a good idea to add a step before this about building the image with
```
./b webots build webotsrobocup
```

   This will start all six robots, three for each team.

> 5. Adjust the view in Webots as that the whole field is visible.

Reviewer 3: ```suggestion
5. Adjust the view in Webots so that the whole field is visible.
```

6. Click the HTML5 animation record button in Webots, as shown in the image below. This will record an interactive version of the game.

>    ![Screenshot of webots emphasising the HTML5 animation record button.](./images/webots-record.png 'creenshot of webots emphasising the HTML5 animation record button.')

Reviewer: ```suggestion
   ![Screenshot of webots emphasising the HTML5 animation record button.](./images/webots-record.png 'screenshot of webots emphasising the HTML5 animation record button.')
```

> 7. Click the button directly to the left of the HTML5 animation button to simulatiously record a standard video of the game.

Reviewer: ```suggestion
7. Click the button directly to the left of the HTML5 animation button to simultaneously record a standard video of the game.
```

8. Once the game has finished, press <kbd>Esc</kbd> to exit fullscreen mode and when prompted save the video. This video can be shared in the Google Drive or on our YouTube channel.

> 9. Click the HTML5 animation record button again to stop the recording and when prompted save the recording. This can be uploaded to the Google Drive and can be used to review the game with more flexibility than a standard video.

Reviewer: How do we  review HTML5 games? With the browser?

Author: Ah yeah good point I'll add in instructions for this.

2. Set up the game configuration.

   1. Open the [game.json file](https://github.com/NUbots/NUWebots/blob/main/webots/projects/samples/contests/robocup/controllers/referee/game.json).
   2. Change the `"blue":` `"id"` to `13`.
   3. Change the `"blue":` `"config"` field to `"team_1.json"`.

3. Open a terminal and run the following to open Webots with the Robocup world and the GameController:

   ```bash
   $WEBOTS_HOME/webots $WEBOTS_HOME/projects/samples/contests/robocup/worlds/robocup.wbt
   ```

   Pause the simulation when it opens. The world and robots will load in but the game will not start yet.

4. Open a separate terminal and build the robot code.

   ```bash
   ./b webots build webotsrobocup
   ```

5. Run the robots

   ```bash
   ./b webots game -r 6 webotsrobocup
   ```

   This will start all six robots, three for each team.

6. Adjust the view in Webots so that the whole field is visible.

7. Click the HTML5 animation record button in Webots, as shown in the image below. This will record an interactive version of the game.

>    ![Screenshot of webots emphasising the HTML5 animation record button.](./images/webots-record.png 'screenshot of Webots emphasising the HTML5 animation record button.')

Reviewer 3: This seems an odd caption to me... Maybe something like this instead?
```suggestion
   ![Screenshot of webots emphasising the HTML5 animation record button.](./images/webots-record.png 'HTML5 animation record button in Webots')
```
